### PR TITLE
Juniper post enum module

### PR DESCRIPTION
**Pull request by h00die**
This is a post module to enumerate configuration items from Juniper devices, similar to the Cisco one that exists.

This is a WIP, as the SSH sessions to ScreenOS devices are being debugged via #9519 .

Do you have a Juniper?  Please test this out!  I only have 2 devices, and they are ones I bought to write this module (non-production), so I'm unaware of newer OS changes to commands/output, or other fields that may be good to pull/save.  Please post censored output and let me know what device and OS you tested against so we can add it to the documentation.

Requires #9524 and https://github.com/rapid7/metasploit-credential/pull/129

## Verification

- [x] Get a shell on a juniper junos/screenos at different priv levels (on JunOS: root=ssh shell, user=cli)
- [x] `use post/juniper/gather/enum_juniper`
- [x] `set session [x]`
- [x] `set verbose true`
- [x] `run`
- [x] **Verify** you get lots of pretty creds

If you want to take a look at the config on the device I tested, you can see it [here](https://github.com/h00die/MSF-Testing-Scripts/blob/master/juniper_ex2200.config) and a no privileges account [here](https://github.com/h00die/MSF-Testing-Scripts/blob/master/juniper_ex2200_access_denied.config)

Check docs for output from my device and what fields are pulled/saved.

### CODE REVIEW

**Comment by h00die** [comment]
Removed WIP, more like ready for review and testing by others.

**Comment by h00die** [comment]
Before I mirror this process for juniper, any feedback?

**Comment by h00die** [comment]
`s/juniper/brocade/`

File: lib/msf/core/auxiliary/juniper.rb [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+      cred[:service_name] = 'radius'
+      create_credential_and_login(cred)
+    end
+
+    config.scan(/pap {[\s]+local-name "(?<ppp_username>.+)";[\s]+local-password "(?<ppp_hash>[^"]+)";/i).each do |result|
+      ppp_username = result[0].strip
+      ppp_hash = result[1].strip
+      print_good("PPTP username #{ppp_username} hash #{ppp_hash} via PAP")
+      cred = credential_data.dup
+      cred[:username] = ppp_username
```
Comment (busterb) [inline lib/msf/core/auxiliary/juniper.rb]: in hindsight, does this need an address too?

Comment (busterb) [reply lib/msf/core/auxiliary/juniper.rb]: oh never mind, this was good (it's set above)

**Comment by busterb** [comment]
Some of the code patterns made me think 'stripping after splitting the config fields is tedious'.

**Comment by tdoan-r7** [comment]
# Release Notes

The post/juniper/gather/enum_juniper.rb module has been added to the framework. It enumerates configuration items from Juniper devices, similar to the Cisco one that exists.

**Pull request merged**